### PR TITLE
Html email

### PR DESCRIPTION
**Pull request by smc395**
Sends an HTML email to users who register in the system.

**Significant Changes**
1. HTML email
2. Activation link is based on environment
3. Clicking on activation link brings them to a page on the Frontend that tells them if their account was successfully activated or not.

**How to Test**
1. Go to register page
2. Register for an account
3. Check email is HTML email
4. Check activation link reflects environment
5. Click on link and check that you see a happy face
6. Sign in with registered user
7. Copy past link into browser and remove part of the activation id
8. Check that you see a sad face
9. Throttle network speed and check that when activation link is clicked a loading screen is given.

**Notes**
:cactus:

### CODE REVIEW

File: Backend/mentii/user_ctrl.py [hunk 1]
```diff
@@ -10,7 +11,7 @@
 from flask import g
 
 
-def register(jsonData, mailer, dbInstance):
+def register(httpOrigin, jsonData, mailer, dbInstance):
```
Comment (aaronmsegal) [inline Backend/mentii/user_ctrl.py]: Should this be refactored to have a single return?

File: Backend/mentii/user_ctrl.py [hunk 2]
```diff
@@ -77,7 +76,7 @@ def isEmailValid(email):
 def isPasswordValid(password):
   return len(password) >= 8
 
-def addUserAndSendEmail(email, password, mailer, dbInstance):
+def addUserAndSendEmail(httpOrigin, email, password, mailer, dbInstance):
```
Comment (aaronmsegal) [inline Backend/mentii/user_ctrl.py]: Should this also be refactored to have a single return?

File: Backend/tests/test_user_ctrl.py [hunk 3]
```diff
@@ -70,7 +70,8 @@ def test_register_fail(self):
     print("Running register fail case Test")
     jsonData = {"email" : "mail@mentii.me"}
     dbInstance = "blah"
-    self.assertNotEqual(usr.register(jsonData,mail,dbInstance), None)
+    httpOrigin = "localhost:2222"
+    self.assertNotEqual(usr.register(httpOrigin,jsonData,mail,dbInstance), None)
```
Comment (aaronmsegal) [inline Backend/tests/test_user_ctrl.py]: assertIsNotNone()

File: Backend/tests/test_user_ctrl.py [hunk 4]
```diff
@@ -121,10 +122,10 @@ def test_addUserAndSendEmail(self):
 
     email = "email@mentii.me"
     password = "password8"
-
+    httpOrigin = "localhost:7777"
```
Comment (aaronmsegal) [inline Backend/tests/test_user_ctrl.py]: double quotes

File: Backend/tests/test_user_ctrl.py [hunk 5]
```diff
@@ -142,9 +143,10 @@ def test_register(self):
 
     email = "mail@mentii.me"
     password = "password"
-
     jsonData = {"email" : email, "password" : password}
-    response = usr.register(jsonData,mail,dynamodb)
+    httpOrigin = "localhost:9999"
```
Comment (aaronmsegal) [inline Backend/tests/test_user_ctrl.py]: double quotes

File: Backend/app.py [hunk 6]
```diff
@@ -91,7 +91,9 @@ def register():
     logger.info(str(flaskResponse))
     return flaskResponse
   dynamoDBInstance = getDatabaseClient()
-  res = user_ctrl.register(request.json, mail, dynamoDBInstance)
+  httpOrigin = request.environ.get('HTTP_ORIGIN')
+  jsonData = request.json
+  res = user_ctrl.register(httpOrigin, jsonData, mail, dynamoDBInstance)
```
Comment (RyanYoung25) [inline Backend/app.py]: Why is the request.json being put in a variable here. No need to change, just curious.

Comment (smc395) [inline Backend/app.py]: readability sake

File: Backend/mentii/user_ctrl.py (hunk 1 above)
Comment (smc395) [inline Backend/mentii/user_ctrl.py]: Addressed

File: Backend/mentii/user_ctrl.py (hunk 2 above)
Comment (smc395) [inline Backend/mentii/user_ctrl.py]: Addressed

**Comment by jonathanmcdaniel** [comment]
@smc395 version number is out of date

**Pull request merged**